### PR TITLE
[FIX] fixing mean molecular weight for different cooling calls

### PR DESCRIPTION
**Pull request by RitaliG**
Hi Alankar,

It might be helpful to include a few changes in `init.c`, where mu is accessed in a customized way when cooling is set to NO, TABULATED or TOWNSEND. Just a bug fix in case one uses a different cooling method, like POWER_LAW.

### CODE REVIEW

**Comment by RitaliG** [comment]
also 'rByrInj' is nowhere used, so that variable can be gotten rid of in the Analysis function

File: init.c [hunk 1]
```diff
@@ -204,7 +219,7 @@ void Analysis (const Data *d, Grid *grid)
     mass_cloud_all[i] = 0.;
   }
 
-  double dV, rByrInj, rho_wind, T_wind, T_gas;
+  double dV, rho_wind, T_wind, T_gas;
```
Comment (dutta-alankar) [inline init.c:222]: Why was this line changed?

File: init.c [hunk 2]
```diff
@@ -128,8 +137,14 @@ void Analysis (const Data *d, Grid *grid)
...
     first = 1;
+
+    #if COOLING==NO || COOLING==TABULATED || COOLING==TOWNSEND 
     double oth_mu[4];
-    mu   = MeanMolecularWeight((double*)d->Vc, oth_mu);
+    double mu = MeanMolecularWeight((double*)d->Vc, oth_mu);
+    #else
+    double mu = MeanMolecularWeight((double*)d->Vc);
+    #endif
+
```
Comment (dutta-alankar) [inline init.c:147]: I agree. Thanks for the update!

**Comment by dutta-alankar** [comment]
@RitaliG Merging. Good catch!
